saleor/tests/e2e/checkout/test_checkout_create_from_order.py

```diff
@@ -0,0 +1,162 @@
+from ..warehouse.utils import create_warehouse
+from .utils import checkout_create_from_order
+
+
+def prepare_product(
+    e2e_staff_api_client,
+    permission_manage_products,
+    permission_manage_channels,
+    permission_manage_shipping,
+    permission_manage_product_types_and_attributes,
+    permission_manage_orders,
+    permission_manage_checkouts,
+    channel_slug,
+):
+    permissions = [
+        permission_manage_products,
+        permission_manage_channels,
+        permission_manage_shipping,
+        permission_manage_product_types_and_attributes,
+        permission_manage_orders,
+        permission_manage_checkouts,
+    ]
+    assign_permissions(e2e_staff_api_client, permissions)
+
+    warehouse_data = create_warehouse(e2e_staff_api_client)
+    warehouse_id = warehouse_data["id"]
+
+    warehouse_ids = [warehouse_id]
+    channel_data = create_channel(
+        e2e_staff_api_client, slug=channel_slug, warehouse_ids=warehouse_ids
+    )
+    channel_id = channel_data["id"]
+
+    channel_ids = [channel_id]
+    shipping_zone_data = create_shipping_zone(
+        e2e_staff_api_client,
+        warehouse_ids=warehouse_ids,
+        channel_ids=channel_ids,
+    )
+    shipping_zone_id = shipping_zone_data["id"]
+
+    shipping_method_data = create_shipping_method(
+        e2e_staff_api_client, shipping_zone_id
+    )
+    shipping_method_id = shipping_method_data["id"]
+
+    create_shipping_method_channel_listing(
+        e2e_staff_api_client, shipping_method_id, channel_id
+    )
+
+    product_type_data = create_product_type(
+        e2e_staff_api_client,
+    )
+    product_type_id = product_type_data["id"]
+
+    category_data = create_category(e2e_staff_api_client)
+    category_id = category_data["id"]
+
+    product_data = create_product(e2e_staff_api_client, product_type_id, category_id)
+    product_id = product_data["id"]
+
+    create_product_channel_listing(e2e_staff_api_client, product_id, channel_id)
+
+    stocks = [
+        {
+            "warehouse": warehouse_id,
+            "quantity": 5,
+        }
+    ]
+    product_variant_data = create_product_variant(
+        e2e_staff_api_client,
+        product_id,
+        stocks=stocks,
+    )
+    product_variant_id = product_variant_data["id"]
+
+    create_product_variant_channel_listing(
+        e2e_staff_api_client,
+        product_variant_id,
+        channel_id,
+    )
+    data = draft_order_create(
+        e2e_staff_api_client,
+        channel_id,
+    )
+
+    order_id = data["order"]["id"]
+    order_lines = [{"variantId": product_variant_id, "quantity": 1, "price": 100}]
+    order_data = order_lines_create(e2e_staff_api_client, order_id, order_lines)
+    order_product_variant_id = order_data["order"]["lines"][0]["variant"]
+    order_product_quantity = order_data["order"]["lines"][0]["quantity"]
+
+    return (
+        order_id,
+        order_product_variant_id,
+        order_product_quantity,
+    )
```

Comment:
@fowczarek I need your insight on this - since we do not use utils in any of our e2e tests is it good to start doing that now or should we deal with it later on?

Reply:
We have unique data prepared(until) for each test case. I decided to have test case implementation and data preparing in one file instead of having 2 separate files for each test case(one for the test case and one for data preparation).